### DISCUSSION TITLE
기능에 대해 문의드립니다!

### DISCUSSION THREAD

**Discussion by rjc90** [naverz/zepeto-studio-kor#411] [5. Q&A]
안녕하세요. 컴퓨터 언어란 아무것도 모르는 백지상태의 유저입니다.

피자치즈님 글과 커뮤니티에 올라오는 글보면서 혼자 해보는중인데, 문득 궁금한게 있어서 문의드립니다!

사진처럼 모니터 앞에 서면, 버튼이 나오게는 했습니다!

1.   (1)사진과 같이 앞에서서 버튼을 눌렀을 때, (2)사진과 같이 'B' 와 '1' 그리고 '화살표' 에 색상 강조 표시를 주고 싶습니다. 어떻게 하면 좋을까용.. 비슷한거 참고 할 만한 예시 글이라도 있을까요?

**1번사진** 
![image](https://user-images.githubusercontent.com/29249173/154920392-057a589c-7c85-45af-a3c3-282b58cd5fb0.png)

**2번사진**

![image](https://user-images.githubusercontent.com/29249173/154920317-2a586dfe-384d-427b-a1e1-f7dbed8add05.png)

**Comment by minjujuu** [comment]
강조를 주려는 오브젝트가 3D 오브젝트 같은데 맞을까요?
맞다면 코드는 다음과 같이 작성해볼 수 있을 거 같아요!

```
import { GameObject, Input, Material, Renderer } from 'UnityEngine'
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class MaterialTest extends ZepetoScriptBehaviour {

    public cube:GameObject;
    public material_red:Material;

    Update() {
        if(Input.GetMouseButtonDown(0))
            this.cube.GetComponent<Renderer>().material.color = Color.blue;  // Color 이용 방식

        if (Input.GetMouseButtonDown(1))
            this.cube.GetComponent<Renderer>().material = this.material_red;    // Material 교체 방식

    }

}
```

인스펙터 창에서는 사진과 같이 연결해주었습니다.
![image](https://user-images.githubusercontent.com/79794414/154933463-10611664-2554-410d-a323-d78564f5af8a.png)

그리고 material을 변경할 경우 오브젝트에는 Renderer가 있어야 합니다.
![image](https://user-images.githubusercontent.com/79794414/154932424-1bbb1535-4897-4230-a814-dadccc887010.png)

버튼을 누르면 색상 강조 표시를 하기 위해서는 UI Event를 이용하여 onClick 리스너를 등록해주면 될 것 같습니다.


```
Start()
{
    this.btnUI.onClick.AddListener(()=>{
        // 강조 코드 예시 
        this.cube.GetComponent<Renderer>().material.color = Color.blue;  // Color 이용 방식
    });
}
```
참고 : https://studio.zepeto.me/guides/ui-events

**Reply by rjc90** [reply]
감사합니다! 한번 따라서 해보겠습니다!!

**Reply by rjc90** [reply]
감사합니다! 작동 확인하였습니다! ㅎㅎㅎ
